sale_stock_line_customer_ref/tests/test_customer_ref.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo.tests.common import Form, SavepointCase
+
+
+class TestSaleLineCustomerRef(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.wh = cls.env.ref("stock.warehouse0")
+        cls.wh.delivery_steps = "pick_pack_ship"
+        cls.order = cls._create_order()
+        cls.order.action_confirm()
+
+    @classmethod
+    def _create_order(cls):
+        # Create a SO with two lines sharing the same product but two
+        # different customer references
+        with Form(cls.env["sale.order"]) as form:
+            form.partner_id = cls.env.ref("base.res_partner_1")
+            for i in range(2):
+                with form.order_line.new() as line:
+                    line.product_id = cls.env.ref("product.consu_delivery_01")
+                    line.customer_ref = f"TEST_{i}"
+            order = form.save()
+            return order
+
+    def test_customer_ref(self):
+        # Ensure we have two moves that didn't get merged despite the identical
+        # product as their customer reference is different.
+        self.assertEqual(len(self.order.order_line.move_ids), 2)
```

Comment:
I would add more than 1 record per case

Reply:
You mean, for the same product, put some SO lines with the same customer ref, and others with a different one? I'll do that.

Reply:
I pushed a fixup that extend the tests and add some comments in the code as I wrote this initially without considering some obvious behavior of moves merging.